statemachine/factory.py

```diff
@@ -89,12 +97,41 @@ def _check_final_states(cls):
                 ).format([s.id for s in final_state_with_invalid_transitions])
             )
 
+    def _check_trap_states(cls):
+        trap_states = [s for s in cls.states if not s.final and not s.transitions]
+        if trap_states:
+            message = _(
+                "All non-final states should have at least one outgoing transition. "
+                "These states have no outgoing transition: {!r}"
+            ).format([s.id for s in trap_states])
+            if cls._strict_states:
+                raise InvalidDefinition(message)
+            else:
+                warnings.warn(message, UserWarning, stacklevel=1)
+
+    def _check_reachable_final_states(cls):
+        if any(s.final for s in cls.states):
+            disconnected_states = [
+                state for state in cls.states if not state.final 
+                and not any(s.final for s in visit_connected_states(state))
+            ]
```

Comment:
**suggestion:** Extract the search algo to a method.

```suggestion
            disconnected_states = cls._states_without_path_to_final_states()
```

Reply:
I moved the logic to a method in the metaclass. I don't think the logic is very useful outside of this one check